snuba/migrations/system/0001_migrations.py

```diff
@@ -0,0 +1,29 @@
+from typing import Sequence
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migrations
+from snuba.migrations import operations
+
+
+class Migration(migrations.Migration):
+    is_dangerous = False
+    dependencies = None
+
+    def forwards_local(self) -> Sequence[operations.Operation]:
+        return [
+            operations.RunSql(
+                storage_set=StorageSetKey.MIGRATIONS,
+                statement="""\
+                    CREATE TABLE IF NOT EXISTS migrations_local (app String,
+                    migration_id String, timestamp DateTime, status Enum('completed' = 0,
+                    'in_progress' = 1), consumer_offsets String)
+                    ENGINE = ReplacingMergeTree() ORDER BY (app, migration_id);
```

Comment:
You need to provide a version column for a replacing merge tree to ensure the most recent state is the one that wins when merging happens
https://clickhouse.tech/docs/en/engines/table-engines/mergetree-family/replacingmergetree/#creating-a-table

Reply:
According to that doc it will be the `Last in the selection, if ver not set.` Isn't that the same as most recent state?

Reply:
last in the selection does not mean last inserted in order of time. The order by clause does not include the timestamp. It may work but I don't think you have this guarantee, to err on the safe side I would intoduce a logical version so that none needs to try to figure out whether this works by an implementation detail of clickhouse.

Reply:
Ok, updated.